x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/snapshot_policies/snapshot_policies.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment } from 'react';
+
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+
+import {
+  EuiButtonIcon,
+  EuiCallOut,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiSpacer,
+} from '@elastic/eui';
+
+import { SnapshotPoliciesProps } from './';
+
+import { useLoadSnapshotPolicies } from '../../../../services/api';
+
+export const SnapshotPolicies: React.FunctionComponent<SnapshotPoliciesProps> = ({
+  value,
+  onChange,
+}) => {
+  const { error, isLoading, data, sendRequest } = useLoadSnapshotPolicies();
+
+  const policies = data.map((name: string) => ({
+    label: name,
+    value: name,
+  }));
+
+  const onComboChange = (options: EuiComboBoxOptionOption[]) => {
+    if (options.length > 0) {
+      onChange(options[0].label);
+    } else {
+      onChange('');
+    }
+  };
+
+  const onCreateOption = (newValue: string) => {
+    onChange(newValue);
+  };
+
+  let calloutContent;
+  if (error) {
+    calloutContent = (
+      <Fragment>
+        <EuiSpacer size="m" />
+        <EuiCallOut
+          data-test-subj="policiesErrorCallout"
+          size="s"
+          iconType="help"
+          color="warning"
+          title={
+            <Fragment>
+              <FormattedMessage
+                id="xpack.indexLifecycleMgmt.editPolicy.deletePhase.noPoliciesLoadedTitle"
+                defaultMessage="Couldn't load existing policies."
+              />
+
+              <EuiButtonIcon
+                size="s"
+                color="warning"
+                onClick={sendRequest}
+                iconType="refresh"
+                aria-label={i18n.translate(
+                  'xpack.indexLifecycleMgmt.editPolicy.deletePhase.reloadPoliciesLabel',
+                  {
+                    defaultMessage: 'Reload policies',
+                  }
+                )}
+              />
+            </Fragment>
+          }
+        >
+          <FormattedMessage
+            id="xpack.indexLifecycleMgmt.editPolicy.deletePhase.noPoliciesLoadedMessage"
+            defaultMessage="You still can type in a policy name."
```

Comment:
If we do decide to keep these I think we should get a review from esdocs@

Reply:
👍